tycho-pomgenerator-plugin/src/main/java/org/eclipse/tycho/pomgenerator/GenerateDependencyPomsMojo.java

```diff
@@ -0,0 +1,201 @@
+     * directory.
+     */
+    @Parameter(property = "forceGeneration", defaultValue = "false")
+    private boolean forceGeneration;
+
+    /**
+     * Generates the pom for the current reactor project in case it is an Eclipse feature or (test-)
+     * plugin.
+     * 
+     * @throws MojoExecutionException
+     */
+    @Override
+    public void execute() throws MojoExecutionException {
+        synchronized (LOCK) {
+            switch (project.getPackaging()) {
+            case TYPE_ECLIPSE_PLUGIN:
+                generatePom("test-plugin-pom.xml");
+                break;
+            case TYPE_ECLIPSE_TEST_PLUGIN:
+                generatePom("plugin-pom.xml");
+                break;
+            case TYPE_ECLIPSE_FEATURE:
+                generatePom("feature-pom.xml");
+                break;
+            default:
+                // ignore
+            }
+        }
+    }
+
+    /**
+     * Generates the enhanced pom.xml for the current reactor project. The file itself is stored
+     * under "/target/generated-pom/pom.xml" and contains a list of all direct dependencies.<br>
+     * <b>NOTE:</b> Handwritten poms take precedence. So in case the program detects that such a
+     * file already exists in the base directory, the generation process is skipped and the existing
+     * file is used instead. This behavior can be overwritten by setting the {@code forceGeneration}
+     * flag.
+     * 
+     * @throws MojoExecutionException
+     */
+    private void generatePom(String templateName) throws MojoExecutionException {
+        Path baseDir = project.getBasedir().toPath();
+        Path outDir = baseDir.resolve("target/generated-pom");
+        Path pom = baseDir.resolve("pom.xml");
+
+        if (Files.exists(pom) && !forceGeneration) {
+            getLog().debug("pom.xml already exists. Ignore...");
+            return;
+        }
+
+        Model model = readPomTemplate(templateName);
+        pom = outDir.resolve("pom.xml");
+        try {
+            Files.createDirectories(pom.getParent());
+        } catch (IOException e) {
+            throw new MojoExecutionException(e.getMessage(), e);
+        }
+
+        model.setGroupId(project.getGroupId());
+        model.setArtifactId(project.getArtifactId());
+        model.setVersion(project.getVersion());
+        model.setDependencies(getMinimalReactorDependencies());
+
+        getLog().info("Write generated pom.xml to " + pom.toFile());
+        // Write to disc
+        writePom(outDir.toFile(), model);
+        // Update the path of the pom.xml in the project
+        project.setPomFile(pom.toFile());
+    }
+
+    /**
+     * Calculates the direct dependencies of the current reactor project. I.e. the minimal amount of
+     * dependencies required to include all (transitive) dependencies.
+     *
+     * #TODO project.getArtifacts() should be used instead. However, it returns an empty set, even
```

Comment:
As dependency:list is working correctly as @mickaelistria has mentioned, you might want to take a look how it is done there?

Reply:
by the way its

```
List<Dependency> list = Objects.requireNonNullElse(mavenProject.getDependencies(),
						Collections.emptyList());
```

what you are most probably want to use instead of `project.getArtifacts()` that is only filled if you resolve the project through maven I guess.